libraries/standard/mqtt/utest/mqtt_utest.c

```diff
@@ -78,3 +245,354 @@ void test_MQTT_Init_Invalid_params( void )
     mqttStatus = MQTT_Init( &context, &transport, &callbacks, NULL );
     TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
 }
+
+/* ========================================================================== */
+
+/**
+ * @brief Test MQTT_Connect, except for receiving the CONNACK.
```

Comment:
Yes, the happy path does seem out of place. I removed the happy path case and made it its own function, but left the rest of the test as is. I didn't see value in separating them further, since the test cases are all similar in that they're error paths that occur before CONNACK reception